test/tokenstore.hs

```diff
@@ -73,13 +88,26 @@ instance Arbitrary Token where
 instance Arbitrary TokenType where
     arbitrary = arbitraryBoundedEnum
 
-instance Arbitrary TokenGrant where
-    arbitrary =
+instance Arbitrary ValidTokenGrant where
+    arbitrary = ValidTokenGrant <$> (
         TokenGrant <$> arbitrary
```

Comment:
If you just need different TokenGrant Arbitrary instances here, I'd have gone with a newtype wrapper, better potential code re-use.